src/components/dock/dock-item/dock-item.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    h,
+    Prop,
+    State,
+} from '@stencil/core';
+import { DockItem } from '../dock.types';
+import { createRandomString } from '../../../util/random-string';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'limel-dock-item',
+    shadow: false,
+    styleUrl: 'dock-item.scss',
+})
+
+// buttons in a menu component
+export class DockItemMenu {
```

Comment:
Hmm… I'm having some trouble following your thoughts here 🙂

The tag for this component is `limel-dock-item`, but the class name is `DockItemMenu`. I didn't pay that much mind at first, as I just assumed that you had renamed the component and forgotten the class name or something, which I do myself all the time.

But then I realised that there's an interface called `DockItem` and this component does _not_ implement that interface, but has a property called `item` that expects an object of type `DockItem` 😮

I feel very confused right now… 😕

Reply:
I was also very confused here, expecting a menu that should be passed items as a prop, not a menu item passed a single item along with an optional component prop that could implement a menu...

Reply:
The interface should be `DockItemType`

Reply:
⚡️

Reply:
> It's been 10 months since we started playing with this component.

😮